hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/Options.java

```diff
@@ -518,4 +518,28 @@ public String toString() {
     MD5MD5CRC,  // MD5 of block checksums, which are MD5 over chunk CRCs
     COMPOSITE_CRC  // Block/chunk-independent composite CRC
   }
+
+  /**
+   * Enum to support the varargs for swapBlockList() options.
+   */
+  public enum SwapBlockList {
+    NONE((byte) 0), // No options, swap the block list.
+    ONE_WAY_BLOCK_SWAP((byte) 1), // Skip replacing source with dst info.
```

Comment:
What is the use case for one way swap? My understanding is that at the end of one way swap, both files will have the same block list. I think that will be incorrect as NameNode assumes 1:1 correspondence between blocks and files.